### PR TITLE
cleanup some benchmark related code

### PR DESCRIPTION
**Pull request by DavidKorczynski**
While working on https://github.com/google/oss-fuzz-gen/issues/494 I
need to adjust a few things wrt. Benchmarks, as the test-to-harness
logic doesn't fit the benchmark abstraction entirely. This commit is a
step towards simplifying some of the code around benchmarks. I also
removed some code that was no longer used, e.g. `manual_fix` in
`run_one_experiment.py`.

### CODE REVIEW

**Comment by DavidKorczynski** [comment]
/gcbrun exp -n dk-5431 -m vertex_ai_gemini-1-5 -b comparison

**Comment by DavidKorczynski** [comment]
Experiment is looking good

**Review by oliverchang — APPROVED** [review]

File: llm_toolkit/prompt_builder.py [hunk 1]
```diff
@@ -115,9 +112,11 @@ class DefaultTemplateBuilder(PromptBuilder):
 
   def __init__(self,
                model: models.LLM,
+               benchmark: Optional[Benchmark] = None,
                template_dir: str = DEFAULT_TEMPLATE_DIR):
     super().__init__(model)
     self._template_dir = template_dir
+    self.benchmark = benchmark
```
Comment (DonggeLiu) [inline llm_toolkit/prompt_builder.py:119]: Thanks @DavidKorczynski ! I also wanted to make this change for sooo long, it's great that you did it!

File: run_one_experiment.py [hunk 2]
```diff
@@ -213,13 +213,73 @@ def prepare(oss_fuzz_dir: str) -> None:
...
+    # For Java projects
+    builder = prompt_builder.DefaultJvmTemplateBuilder(model, benchmark,
+                                                       template_dir)
+  else:
+    if prompt_builder_to_use == 'CSpecific':
+      builder = prompt_builder.CSpecificBuilder(model, benchmark, template_dir)
+    else:
+      # Use default
+      builder = prompt_builder.DefaultTemplateBuilder(model, benchmark,
+                                                      template_dir)
```
Comment (DonggeLiu) [inline run_one_experiment.py:257]: nit:
```python
if benchmark.language == 'jvm':
  ...
elif prompt_builder_to_use == 'CSpecific':
  ...
else:
  ...
```

**Review by DonggeLiu — COMMENTED** [review]
LGTM

**Pull request merged**